replica/table.cc

```diff
@@ -1222,8 +1222,7 @@ future<std::unordered_set<sstring>> table::get_sstables_by_partition_key(const s
 }
 
 const sstables::sstable_set& table::get_sstable_set() const {
-    // main sstables is enough for the outside world. sstables in other set like maintenance is not needed even for expiration purposes in compaction
-    return *_compaction_group->main_sstables();
+    return *_sstables;
 }
```

Comment:
Are there any current users of this method who might be suprised by the change?

Reply:
Actually, I was worried about current users of this method who might be getting incorrect behavior due to maintanenance set being left out :-) Compaction layer was one of the users and it's now interacting with table through table_state which exposes main sstables only from the compaction group.

Reply:
The change to limit get_sstable_set() to sstables in main set was done after we introduced compound set for off-strategy compaction. Maybe it wasn't the right decision as one could incorrectly assume get_sstable_set() will provide its caller with all sstables (including the ones in maintenance set). But now with table_state() we no longer need to limit this method to main_sstables, but rather return the compound set which spans all the sstables sets in a table.

Reply:
So what's the answer? Will this change cause a bug, or is it fixing a bug, or are all current callers unaffected?

Reply:
Turns out it fixes a problem in table::get_sstables_by_partition_key(), which won't see the sstables in maintenance set until reshape completes. That function is exposed via API only.

Will extract the fix into a separate series, and add an unit test.

I also would like to understand the impact of the change in view_builder::initialize_reader_at_current_token().
@nyh, please help. We do register maintenance sstables (stored in maintenance set) for view update using view_update_generator::register_staging_sstable(), so they're consumed by view_update_generator loop. But view_builder::initialize_reader_at_current_token() uses table::get_sstable_set() to create a local reader for a given current token. What's the impact if that reader misses sstables that are temporarily stored in the table's maintenance set (but which were already registered in view_update_generator as mentioned above)?